spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/catalog/GravitinoCatalogManager.java

```diff
@@ -37,22 +37,18 @@ public class GravitinoCatalogManager {
 
   private volatile boolean isClosed = false;
   private final Cache<String, Catalog> gravitinoCatalogs;
-  private final String metalakeName;
-  private final GravitinoMetalake metalake;
-  private final GravitinoAdminClient gravitinoClient;
+  private final GravitinoClient gravitinoClient;
 
-  private GravitinoCatalogManager(String gravitinoUri, String metalakeName) {
-    this.metalakeName = metalakeName;
-    this.gravitinoClient = GravitinoAdminClient.builder(gravitinoUri).build();
+  private GravitinoCatalogManager(Supplier<GravitinoClient> clientBuilder) {
+    this.gravitinoClient = clientBuilder.get();
     // Will not evict catalog by default
     this.gravitinoCatalogs = CacheBuilder.newBuilder().build();
-    this.metalake = gravitinoClient.loadMetalake(metalakeName);
   }
 
-  public static GravitinoCatalogManager create(String gravitinoUrl, String metalakeName) {
+  public static GravitinoCatalogManager create(Supplier<GravitinoClient> clientBuilder) {
     Preconditions.checkState(
         gravitinoCatalogManager == null, "Should not create duplicate GravitinoCatalogManager");
-    gravitinoCatalogManager = new GravitinoCatalogManager(gravitinoUrl, metalakeName);
+    gravitinoCatalogManager = new GravitinoCatalogManager(clientBuilder);
```

Comment:
Client can't a field. Otherwise, the client won't support UserGroupInformation.doAs.

Reply:
sorry, I couldn't get your point, could describe more details?

Reply:
You can use the code
```
UserGroupInformation rory = UserGroupInformation.getProxyUser(xxxx);
rory.doAs {
    sql("insert XXX into gravitino.xxx.xxx")
}
```
This client user should use rory.

Reply:
Maybe you can create an issue to track the user mapping and user proxy of connector.

Reply:
I don't have a strong background and option to add this, could you add this issue?